### ISSUE TITLE
3.7 on OSX64.ARM: Unable to load shared library 'gdal_wrap' or one of its dependencies.

### ISSUE THREAD

**Issue by bertt** [MaxRev-Dev/gdal.netcore#109]
Hi, I've upgraded my sample https://github.com/bertt/GdalOnNet6Sample/blob/master/Gdal.Console/Program.cs to 3.7, works fine on Windows/Linux but on OSX64ARM - Ventura 13.4 I see a runtime error in ConfigureAll: 

System.DllNotFoundException: Unable to load shared library 'gdal_wrap' or one of its dependencies.

Package MaxRev.Gdal.MacosRuntime.Minimal.arm64 is referenced correctly.

Maybe some other dependency is missing?

**Comment by bertt** [comment]
Same eror on OSX64 https://github.com/bertt/GdalOnNet6Sample/actions/runs/5625989775/job/15245887073#step:6:23

**Comment by MaxRev-Dev** [comment]
It's possible that the CI machine doesn't have the necessary preinstalled libraries. I'm not sure how to recreate this issue. I tested the packages on three different machines and didn't encounter any problems. If you can debug the problem on your local (osx.arm64/osx.x64) machine, it would be helpful to get more information about the missing dependencies.

**Comment by bertt** [comment]
on my local machine I see 'libgdal_warp.dylib' in Debug/runtimes/net6.0/runtimes/osx-arm64/native, maybe thats the one it's searching? Strange it mentions 'gdal_wrap' but I see 'libgdal_wrap.dylib' in the folder.

<img width="771" alt="Screenshot 2023-07-21 at 23 04 19" src="https://github.com/MaxRev-Dev/gdal.netcore/assets/538812/b0d442d0-5fb8-478f-9aaa-810abfbd68c1">

**Comment by bertt** [comment]
tried copying the *.dylib files to the net6.0 folder but same error :-(

**Comment by MaxRev-Dev** [comment]
At first, it tries to open libgdal_wrap. But when it fails - it advances to the next name in LD list.
As it says, you can also specify a DYLD_PRINT_LIBRARIES debug variable to view if something fails.
`DYLD_PRINT_LIBRARIES=1 ./app` 
Also, consider using **otool** to get a list of dependencies for libgdal_wrap and libgdal.33 as well.
`otool -L "/root/gdal.netcore/build-osx/gdal-build/lib/libgdal.dylib"`
And yes, there's no need to copy libraries. You need to find a missing dependency.

**Comment by bertt** [comment]
$  DYLD_PRINT_LIBRARIES=1 ./Gdal.Console 

gives:

```
dyld[60387]: <4EBD88C5-E74D-3BCD-9D7D-D16298084D31> /usr/lib/swift/libswift_RegexParser.dylib
dyld[60387]: <EF3E6C7B-D836-34D9-AB1C-C05561B80669> /usr/local/share/dotnet/x64/shared/Microsoft.NETCore.App/6.0.11/libclrjit.dylib
Unhandled exception. dyld[60387]: <1902DE04-A7B6-3298-A88D-6D4A1C06DC83> /usr/local/share/dotnet/x64/shared/Microsoft.NETCore.App/6.0.11/libSystem.Native.dylib
System.TypeInitializationException: The type initializer for 'OSGeo.GDAL.GdalPINVOKE' threw an exception.
 ---> System.TypeInitializationException: The type initializer for 'SWIGExceptionHelper' threw an exception.
 ---> System.DllNotFoundException: Unable to load shared library 'gdal_wrap' or one of its dependencies. In order to help diagnose loading problems, consider setting the DYLD_PRINT_LIBRARIES environment variable: dlopen(libgdal_wrap, 0x0001): tried: 'libgdal_wrap' (no such file), '/System/Volumes/Preboot/Cryptexes/OSlibgdal_wrap' (no such file), '/usr/lib/libgdal_wrap' (no such file, not in dyld cache), 'libgdal_wrap' (no such file), '/usr/local/lib/libgdal_wrap' (no such file), '/usr/lib/libgdal_wrap' (no such file, not in dyld cache)
   at OSGeo.GDAL.GdalPINVOKE.SWIGExceptionHelper.SWIGRegisterExceptionCallbacks_Gdal(ExceptionDelegate applicationDelegate, ExceptionDelegate arithmeticDelegate, ExceptionDelegate divideByZeroDelegate, ExceptionDelegate indexOutOfRangeDelegate, ExceptionDelegate invalidCastDelegate, ExceptionDelegate invalidOperationDelegate, ExceptionDelegate ioDelegate, ExceptionDelegate nullReferenceDelegate, ExceptionDelegate outOfMemoryDelegate, ExceptionDelegate overflowDelegate, ExceptionDelegate systemExceptionDelegate)
   at OSGeo.GDAL.GdalPINVOKE.SWIGExceptionHelper..cctor()
   --- End of inner exception stack trace ---
   at OSGeo.GDAL.GdalPINVOKE.SWIGExceptionHelper..ctor()
   at OSGeo.GDAL.GdalPINVOKE..cctor()
   --- End of inner exception stack trace ---
   at OSGeo.GDAL.GdalPINVOKE.AllRegister()
   at OSGeo.GDAL.Gdal.AllRegister()
   at MaxRev.Gdal.Core.GdalBase.ConfigureGdalDrivers(String gdalDataFolder)
   at MaxRev.Gdal.Core.GdalBase.ConfigureAll()
   at GdalNetCore.Core.GdalKit..ctor() in /Users/bert/dev/github.com/bertt/GdalOnNet6Sample/Gdal.Core/GdalKit.cs:line 12
   at GdalNetCore.Program.Main(String[] args) in /Users/bert/dev/github.com/bertt/GdalOnNet6Sample/Gdal.Console/Program.cs:line 15
[1]    60387 abort      DYLD_PRINT_LIBRARIES=1 ./Gdal.Console
```

**Comment by MaxRev-Dev** [comment]
Nothing special here, could you try running the otool?

**Comment by bertt** [comment]
```
$ otool -L "./runtimes/osx-arm64/native/libgdal_wrap.dylib"
./runtimes/osx-arm64/native/libgdal_wrap.dylib:
        @rpath/libgdal_wrap.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libgdal.33.dylib (compatibility version 33.0.0, current version 33.3.7)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1500.65.0)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.100.3)
```

But all dependencies can't be found:

```
otool -L "@rpath/libgdal_wrap.dylib"                     
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file: @rpath/libgdal_wrap.dylib (No such file or directory)
```

other sample:

```
otool -L "/usr/lib/libSystem.B.dylib"
error: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic: can't open file: /usr/lib/libSystem.B.dylib (No such file or directory)
```

**Comment by bertt** [comment]
libgdal33:

```
otool -L "./runtimes/osx-arm64/native/libgdal.33.dylib"  
./runtimes/osx-arm64/native/libgdal.33.dylib:
        @loader_path/libgdal.33.dylib (compatibility version 33.0.0, current version 33.3.7)
        /usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.100.3)
        @loader_path/libarchive.13.dylib (compatibility version 20.0.0, current version 20.2.0)
        @loader_path/libodbc.2.dylib (compatibility version 3.0.0, current version 3.0.0)
        @loader_path/libodbcinst.2.dylib (compatibility version 3.0.0, current version 3.0.0)
        @loader_path/libxml2.2.10.3.dylib (compatibility version 2.0.0, current version 2.10.3)
        @loader_path/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        @loader_path/libssl.3.dylib (compatibility version 3.0.0, current version 3.0.0)
        @loader_path/libzstd.1.5.5.dylib (compatibility version 1.0.0, current version 1.5.5)
        @loader_path/liblzma.5.4.1.dylib (compatibility version 5.0.0, current version 5.4.1)
        @loader_path/liblz4.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libblosc.1.21.3.dylib (compatibility version 1.0.0, current version 1.21.3)
        @loader_path/libtiff.6.0.0.dylib (compatibility version 6.0.0, current version 6.0.0)
        @loader_path/libproj.25.9.2.0.dylib (compatibility version 25.0.0, current version 25.9.2)
        @loader_path/libqhull_r.8.0.2.dylib (compatibility version 8.0.0, current version 8.0.2)
        @loader_path/libxerces-c-3.2.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libjpeg.62.3.0.dylib (compatibility version 62.0.0, current version 62.3.0)
        @loader_path/libpng16.16.39.0.dylib (compatibility version 16.0.0, current version 16.39.0)
        @loader_path/libLerc.4.dylib (compatibility version 4.0.0, current version 0.0.0)
        @loader_path/libbrunslienc-c.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libbrunslidec-c.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libgif.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libnetcdf.19.dylib (compatibility version 19.0.0, current version 19.0.0)
        @loader_path/libcfitsio.3.49.dylib (compatibility version 3.0.0, current version 3.49.0)
        @loader_path/libhdf5.310.0.0.dylib (compatibility version 310.0.0, current version 310.0.0)
        @loader_path/libwebp.7.1.6.dylib (compatibility version 9.0.0, current version 9.6.0)
        @loader_path/libsqlite3.dylib (compatibility version 0.0.0, current version 0.0.0)
        @loader_path/libpq.5.15.dylib (compatibility version 5.0.0, current version 5.15.0)
        @loader_path/libopenjp2.2.5.0.dylib (compatibility version 7.0.0, current version 2.5.0)
        @loader_path/libkmlbase.1.3.0.dylib (compatibility version 1.0.0, current version 1.3.0)
        @loader_path/libkmldom.1.3.0.dylib (compatibility version 1.0.0, current version 1.3.0)
        @loader_path/libkmlengine.1.3.0.dylib (compatibility version 1.0.0, current version 1.3.0)
        @loader_path/liburiparser.1.0.30.dylib (compatibility version 1.0.0, current version 1.0.30)
        @loader_path/libz.1.2.13.dylib (compatibility version 1.0.0, current version 1.2.13)
        /usr/lib/libpcre2-8.0.dylib (compatibility version 12.0.0, current version 12.2.0)
        @loader_path/libfreexl.1.dylib (compatibility version 3.0.0, current version 3.0.0)
        @loader_path/libexpat.1.8.10.dylib (compatibility version 10.0.0, current version 10.10.0)
        @loader_path/libgeos_c.1.17.1.dylib (compatibility version 1.0.0, current version 1.17.1)
        @loader_path/libcurl.4.8.0.dylib (compatibility version 4.0.0, current version 4.8.0)
        @loader_path/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
        /System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL (compatibility version 1.0.0, current version 1.0.0)
        @loader_path/libhdf5_hl.310.dylib (compatibility version 311.0.0, current version 311.0.0)
        @loader_path/libhdf5.310.dylib (compatibility version 311.0.0, current version 311.0.0)
        /opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
        @loader_path/libbz2.1.0.8.dylib (compatibility version 1.0.0, current version 1.0.8)
        @loader_path/libcurl.4.dylib (compatibility version 13.0.0, current version 13.0.0)
        @loader_path/libxml2.2.dylib (compatibility version 13.0.0, current version 13.3.0)
        @loader_path/libdf.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        @loader_path/libmfhdf.0.dylib (compatibility version 1.0.0, current version 1.0.0)
        @loader_path/libszip.2.1.dylib (compatibility version 2.1.0, current version 2.1.0)
        /usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1500.65.0)
```

**Comment by MaxRev-Dev** [comment]
I don't see any issues here. It seems like I have the same output for libgdal.33.

<details>
  <summary>otool log</summary>
  
  
```
./libgdal.33.dylib:
	@loader_path/libgdal.33.dylib (compatibility version 33.0.0, current version 33.3.7)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.100.3)
	@loader_path/libarchive.13.dylib (compatibility version 20.0.0, current version 20.2.0)
	@loader_path/libodbc.2.dylib (compatibility version 3.0.0, current version 3.0.0)
	@loader_path/libodbcinst.2.dylib (compatibility version 3.0.0, current version 3.0.0)
	@loader_path/libxml2.2.10.3.dylib (compatibility version 2.0.0, current version 2.10.3)
	@loader_path/libcrypto.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	@loader_path/libssl.3.dylib (compatibility version 3.0.0, current version 3.0.0)
	@loader_path/libzstd.1.5.5.dylib (compatibility version 1.0.0, current version 1.5.5)
	@loader_path/liblzma.5.4.1.dylib (compatibility version 5.0.0, current version 5.4.1)
	@loader_path/liblz4.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/libblosc.1.21.3.dylib (compatibility version 1.0.0, current version 1.21.3)
	@loader_path/libtiff.6.0.0.dylib (compatibility version 6.0.0, current version 6.0.0)
	@loader_path/libproj.25.9.2.0.dylib (compatibility version 25.0.0, current version 25.9.2)
	@loader_path/libqhull_r.8.0.2.dylib (compatibility version 8.0.0, current version 8.0.2)
	@loader_path/libxerces-c-3.2.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/libjpeg.62.3.0.dylib (compatibility version 62.0.0, current version 62.3.0)
	@loader_path/libpng16.16.39.0.dylib (compatibility version 16.0.0, current version 16.39.0)
	@loader_path/libLerc.4.dylib (compatibility version 4.0.0, current version 0.0.0)
	@loader_path/libbrunslienc-c.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/libbrunslidec-c.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/libgif.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/libnetcdf.19.dylib (compatibility version 19.0.0, current version 19.0.0)
	@loader_path/libcfitsio.3.49.dylib (compatibility version 3.0.0, current version 3.49.0)
	@loader_path/libhdf5.310.0.0.dylib (compatibility version 310.0.0, current version 310.0.0)
	@loader_path/libwebp.7.1.6.dylib (compatibility version 9.0.0, current version 9.6.0)
	@loader_path/libsqlite3.dylib (compatibility version 0.0.0, current version 0.0.0)
	@loader_path/libpq.5.15.dylib (compatibility version 5.0.0, current version 5.15.0)
	@loader_path/libopenjp2.2.5.0.dylib (compatibility version 7.0.0, current version 2.5.0)
	@loader_path/libkmlbase.1.3.0.dylib (compatibility version 1.0.0, current version 1.3.0)
	@loader_path/libkmldom.1.3.0.dylib (compatibility version 1.0.0, current version 1.3.0)
	@loader_path/libkmlengine.1.3.0.dylib (compatibility version 1.0.0, current version 1.3.0)
	@loader_path/liburiparser.1.0.30.dylib (compatibility version 1.0.0, current version 1.0.30)
	@loader_path/libz.1.2.13.dylib (compatibility version 1.0.0, current version 1.2.13)
	/usr/lib/libpcre2-8.0.dylib (compatibility version 12.0.0, current version 12.2.0)
	@loader_path/libfreexl.1.dylib (compatibility version 3.0.0, current version 3.0.0)
	@loader_path/libexpat.1.8.10.dylib (compatibility version 10.0.0, current version 10.10.0)
	@loader_path/libgeos_c.1.17.1.dylib (compatibility version 1.0.0, current version 1.17.1)
	@loader_path/libcurl.4.8.0.dylib (compatibility version 4.0.0, current version 4.8.0)
	@loader_path/libiconv.2.dylib (compatibility version 9.0.0, current version 9.1.0)
	/System/Library/Frameworks/OpenCL.framework/Versions/A/OpenCL (compatibility version 1.0.0, current version 1.0.0)
	@loader_path/libhdf5_hl.310.dylib (compatibility version 311.0.0, current version 311.0.0)
	@loader_path/libhdf5.310.dylib (compatibility version 311.0.0, current version 311.0.0)
	/opt/local/lib/libz.1.dylib (compatibility version 1.0.0, current version 1.2.13)
	@loader_path/libbz2.1.0.8.dylib (compatibility version 1.0.0, current version 1.0.8)
	@loader_path/libcurl.4.dylib (compatibility version 13.0.0, current version 13.0.0)
	@loader_path/libxml2.2.dylib (compatibility version 13.0.0, current version 13.3.0)
	@loader_path/libdf.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@loader_path/libmfhdf.0.dylib (compatibility version 1.0.0, current version 1.0.0)
	@loader_path/libszip.2.1.dylib (compatibility version 2.1.0, current version 2.1.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1500.65.0)
```  
</details>

**Comment by MaxRev-Dev** [comment]
TL;DR; possibly incompatible SDK. It should be resolved by MinimumSDKVersion in the upcomming release. Tested [in CI](https://github.com/MaxRev-Dev/gdal.netcore/actions/runs/7605008923/job/20709321776#step:6:54).

I believe this issue is related to the OS version mismatch. The packages were built on MacOS Ventura 13. 
Regarding, that the latest version of a GH runner was 12. The native package (3.7.x) can be incompatible with older systems. 

For the next build of the MacOS packages I tried to set [the min SDK version](https://stackoverflow.com/a/11424966) and it worked for `macos-latest` [in CI](https://github.com/MaxRev-Dev/gdal.netcore/actions/runs/7605008923/job/20709321776#step:6:54). I'll mention this in docs and descriptions of packages. They were not expected to run in CI. For production use, we usually choose Linux (debian or ubuntu) and for personal it's more convenient to develop on laptops and usually they are running on _fresh_ OS.

Theoretically, the packages could be built using the `macos-latest` runner. But it's [too expensive](https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#minute-multipliers) (1-minute x 10). So I won't switch to it. Another concern is that I'm using my own gh dev account to build those. That means I'd have to spend a fortune. This repo is OSS, and I have no profit from it.
